downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -0,0 +1,84 @@
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    public static void processCellularComponents(GKInstance reactionInst) throws Exception {
+        // First retrieve proteins, then build GO annotation
+        for (GKInstance proteinInst : GOAGeneratorUtilities.retrieveProteins(reactionInst)) {
+            GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+            GKInstance speciesInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.species);
+            // Check if the protein has any disqualifying attributes.
+            boolean validProtein = GOAGeneratorUtilities.validateProtein(referenceEntityInst, speciesInst);
+            if (validProtein) {
+                String taxonIdentifier = ((GKInstance) speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+                // For CC annotations, alternateGOCompartments are disqualifying.
+                if (!GOAGeneratorUtilities.excludedMicrobialSpecies(taxonIdentifier)) {
+                    if (!speciesWithAlternateGOCompartment.contains(taxonIdentifier)) {
+                        // For CC, we are looking at ALL proteins in a ReactionlikeEvent
+                        generateGOCellularCompartmentLine(proteinInst, referenceEntityInst, reactionInst, taxonIdentifier);
+                    } else {
+                        logger.info("Species has an alternative GO compartment, skipping GO annotation");
+                    }
+                } else {
+                    logger.info("Protein is from an excluded microbial species, skipping GO annotation");
+                }
+            } else {
+                logger.info("Invalid protein, skipping GO annotation");
+            }
+        }
+    }
+
+    /**
+     * Proteins reaching this part of the script are deemed valid for GO annotation. Retrieves the Cellular Compartment accession associated with this instance and then calls the GOA line generator.
+     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
+     * @param referenceEntityInst -- GKInstance, Reference Entity instance of the protein.
+     * @param reactionInst -- GKInstance, Parent reaction that the protein is associated with.
+     * @param taxonIdentifier -- String, CrossReference ID of protein's species.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static void generateGOCellularCompartmentLine(GKInstance proteinInst, GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        String reactomeIdentifier = GOAGeneratorConstants.REACTOME_IDENTIFIER_PREFIX  + ((GKInstance) reactionInst.getAttributeValue(ReactomeJavaConstants.stableIdentifier)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
```

Comment:
Maybe a helper method would be useful to return the stable identifier value and then that can be joined to the prefix

Reply:
Yep that cleaned this up nicely